docs/concepts/typings.md

```diff
@@ -4,4 +4,103 @@ title: Type hints
+| ------------- | -------------------------- | --- |
+| `ID`          | `strawberry.ID`            |
+| `String`      | `str`                      |
+| `Integer`     | `int`                      |
+| `Float`       | `float`                    |
+| `Decimal`     | `decimal.Decimal`          |
+| `Array`, `[]` | `typing.List` or `list`    |
+| `Union`       | `typing.Union` or `        | `   |
+| `Nullable`    | `typing.Optional` or `None | `   |
+| `date`        | `datetime.date`            |
+| `timetz`      | `datetime.time`            |
+| `timestamptz` | `datetime.datetime`        |
+
+where `typing`, `datetime`, and `decimal` are all part of the Python standard library. There is also `typing.Dict` that possesses no mapping since it is the entire structure of the graphQL query itself that is a dictionary.
+
+There are a few different ways in which these Python type hints can be used to express the required Strawberry graphQL type annotation.
+
+- For versions of Python >= 3.10, it is possible to annotate an array of types with `list[Type]`. However, for all previous versions, `typing.List[Type]` must be used instead.
+- The annotation `|` is shorthand for `typing.Union[]`, allowing either of `typing.Union[TypeA, TypeB]` or `TypeA | TypeB` interchangably.
+- The annotation `typing.Optional[Type]` is shorthand for `typing.Union[None, Type]`, which is itself equivalent to `None | Type`.
+
+## Example
+
+A complete example of this, extending upon [Schema basics](https://strawberry.rocks/docs/general/schema-basics), might be the following:
+
+```python
+import datetime
+import typing
+import strawberry
+
+BOOKS_LOOKUP = {
+    'Frank Herbert': [{
+        'title': 'Dune',
+        'date_published': '1965-08-01',
+        'price': '5.99',
+        'isbn': 9780801950773
+    }],
+}
+
+@strawberry.type
+class Book:
+    title: str
+    author: 'Author'
+    date_published: datetime.date
+    price: decimal.Decimal
+    isbn: Union[int, str] # could be 1234567890000 or '123-4-56-789000-0'
+
+def get_books_by_author(root: 'Author') -> typing.List['Book']:
+    # error will be thrown in here if a book in BOOKS_LOOKUP is missing a field
+    # since none of the types of Book are optional
+    stored_books: typing.List[typing.Dict[str, str | int]] = BOOKS_LOOKUP[root.name]
+    return [Book(
+        title = book.get('title'),
+        author = root,
+        date_published = book.get('date_published'),
+        price = book.get('price'),
+        isbn = book.get('isbn')
+    ) for book in stored_books]
+
+@strawberry.type
+class Author:
+    name: str
+    books: typing.List['Book'] = strawberry.field(resolver=get_books_by_author)
+
+@strawberry.type
+class Group:
+    name: Optional[str] # groups of authors don't necessarily have names
+    authors: typing.List['Author']
+
+    @strawberry.field
+    def books(self) -> typing.List['Book']:
+        books = []
+        for author in self.authors:
+            books += get_books_by_author(author)
```

Comment:
```suggestion
        books = list(map(get_books_by_author, self.authors))
```
Not sure if we want to be more explicit for an example, though

Reply:
I think I'll leave the more explicit version for clarity